fineract-progressive-loan/src/test/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleGeneratorTest.java

```diff
@@ -0,0 +1,141 @@
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class LoanScheduleGeneratorTest {
+
+    private static final ProgressiveEMICalculator emiCalculator = new ProgressiveEMICalculator(null);
+    private static MockedStatic<MoneyHelper> moneyHelper = Mockito.mockStatic(MoneyHelper.class);
+    private static final ApplicationCurrency applicationCurrency = new ApplicationCurrency("USD", "USD", 2, 1, "USD", "$");
+    private static final MonetaryCurrency monetaryCurrency = MonetaryCurrency.fromApplicationCurrency(applicationCurrency);
+
+    @BeforeAll
+    public static void init() {
+        moneyHelper.when(MoneyHelper::getRoundingMode).thenReturn(RoundingMode.HALF_EVEN);
+        moneyHelper.when(MoneyHelper::getMathContext).thenReturn(new MathContext(12, RoundingMode.HALF_EVEN));
+    }
+
+    @AfterAll
+    public static void destruct() {
+        moneyHelper.close();
+    }
+
+    @Test
+    void testGenerateLoanSchedule() {
+        HolidayDetailDTO holidayDetailDTO = null;
+        Set<LoanCharge> loanCharges = Collections.emptySet();
+        List<LoanTermVariationsData> loanTermVariationsDataForTest = createLoanTermVariationsDataForTest();
+
+        LoanRepaymentScheduleModelData modelData = new LoanRepaymentScheduleModelData(LocalDate.of(2024, 1, 1), // schedule
+                                                                                                                // generation
+                                                                                                                // start
+                                                                                                                // date
+                applicationCurrency, // currency
+                BigDecimal.valueOf(192.22), // disbursement amount
+                LocalDate.of(2024, 1, 15), // disbursement date
+                6, // number of repayments
+                1, // repayment frequency
+                "MONTHS", // repayment frequency type
+                BigDecimal.valueOf(9.99), // nominal interest rate
+                true, // downpayment enabled
+                DaysInMonthType.DAYS_30, // daysInMonth
+                DaysInYearType.DAYS_360, // daysInYear
+                null, // downpayment percentage (optional)
+                null, // installmentAmountInMultiplesOf (optional)
+                null, // fixed_length (optional)
+                loanTermVariationsDataForTest, // loanTermVariationsDataForTest
+                holidayDetailDTO, // holidayDetailDTO
+                loanCharges // loanCharges
+        );
+
+        final MathContext mc = MoneyHelper.getMathContext();
+
+        final List<LoanScheduleModelRepaymentPeriod> expectedRepaymentPeriods = new ArrayList<>();
+
+        expectedRepaymentPeriods.add(repayment(1, LocalDate.of(2024, 1, 1), LocalDate.of(2024, 2, 1)));
```

Comment:
This is not really testing anything... would you mind to add a test which tests whether all the calculated amounts are correct?

We should see the correct number of repayment periods were created (you are asserting the number), but all the relevant amounts / balances should be tested as well!